networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -57,9 +60,11 @@ public class ConnectionManager extends Service {
   private final Counter failedConnectionCounter;
   private final PeerPools peerPools = new PeerPools();
   private final Collection<Predicate<DiscoveryPeer>> peerPredicates = new CopyOnWriteArrayList<>();
+  private final AtomicInteger startupCountdownStage = new AtomicInteger(5);
```

Comment:
It may make sense to search more frequently while we have less than some number of peers rather than a fixed number of times? Maybe even just switch to the slower schedule once we have some peers.

Reply:
I'd opt in to keep number of times as we could have networks with small total number of peers such as test networks or even private networks where behavior based on the predefined number of peers could be very buggy

Reply:
I think if you have discovery enabled and no peers then retrying frequently makes sense. So slowing down once we have some peers would seem pretty reasonable.

Reply:
I think we have 3 options:
1. 5 fast retries unconditionally
2. Fast retries until certain number of peers reached. 10? I think even 10 could be not enough for exact subnet but setting more could force it to hit frequently forever
3. So, combining 2 something like: after 50 peers or 3 minutes switch to 30 seconds retries

I did first as it looks simple and works most times. I don't like 2nd. 3rd looks a bit more complex but always good in behavior. Do you think I should do something like 3rd option?

Reply:
Ultimately the problem we're trying to solve is that we take too long to start finding peers because the discovery system hasn't had time to start filling tables before the first time we ask it for peers to try to connect to, and then we don't ask for peers again for another 30 seconds (there's that slightly odd second attempt immediately after the first but it hits the same problem that discovery hasn't received any responses yet).

So we really only need to try rapidly until we find the first peer, at which point we know discovery has found some peers. You can see from the logs in the description, we're not finding many more peers between each rapid search after that first batch, so we're not gaining much by continuing to run the search so often and can back off and just run at the usual rate.  If we keep running discovery rapidly for too long, we burn a fair bit of extra CPU and could wind up getting ignored by other nodes for requesting neighbours too often.

That's why I think something really simple like retry fast until we find the first peer makes sense. Nice and simple and solves the slow start problem we're actually targeting.